packages/cli-types/src/index.ts

```diff
@@ -134,17 +136,15 @@ export interface Config {
     platforms: Array<string>;
     providesModuleNodeModules: Array<string>;
   };
-}
-
-type Diff<T, U> = T extends U ? never : T;
+};
 
 /**
- * Shares some structure with ConfigT, except that haste and root
+ * Shares some structure with Config, except that haste and root
  * are calculated and can't be defined
  */
-export type UserConfig = Diff<Config, {root: any; haste: any}> & {
-  reactNativePath: string | void;
 
+export type UserConfig = Omit<Omit<Config, 'root'>, 'haste'> & {
```

Comment:
I think this can be replaced by: `Omit<Config, 'root' | 'haste'>`

Reply:
Correct. When I gave him the example, I typed it like that by accident. My bad 😅. I updated the playground example up top the other week to reflect that.

Reply:
My bad, fixed :)